### PR TITLE
Fix transaction panics in endTransaction with "transaction not found"

### PR DESCRIPTION
**Pull request by dsplayerX**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Transaction Concurrency Test available. This test assesses the concurrency of isolated tasks executing transactions concurrently, verifying they complete a specified number of transactions successfully.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: transaction-ballerina/tests/transaction_concurrency.bal [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    function init(string name) {
+        self.name = name;
+    }
+
+    public isolated function execute() {
+        int count = 0;
+        lock {
+            count = self.counter.cloneReadOnly();
+            if (count >= 100) {
```
Comment (SasinduDilshara) [inline transaction-ballerina/tests/transaction_concurrency.bal:42]:
```suggestion
            if count >= 100 {
```
In ballerina the best practice is not to use brackets in the if expression.

File: transaction-ballerina/tests/transaction_concurrency.bal [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                return;
+            }
+            self.counter += 1;
+        }
+
+        var err = trap performTransaction();
+
+        if !(err is ()) {
+            test:assertFail("Error in task " + self.name + " : " + err.toString());
```
Comment (SasinduDilshara) [inline transaction-ballerina/tests/transaction_concurrency.bal:51]: shall we use string templates in here

File: transaction-ballerina/tests/transaction_concurrency.bal [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        lock {
+            int i = taskCounterMap.get(self.name);
+            taskCounterMap[self.name] = (i + 1);
+        }
+    }
+
+    public isolated function scheduleTaskExecution(decimal interval) {
+        do {
+            var _ = check task:scheduleJobRecurByFrequency(self, interval);
```
Comment (SasinduDilshara) [inline transaction-ballerina/tests/transaction_concurrency.bal:62]:
```suggestion
            check task:scheduleJobRecurByFrequency(self, interval);
```

File: transaction-ballerina/tests/transaction_concurrency.bal [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    var taskD = new TaskExecuter("D");
+    var taskE = new TaskExecuter("E");
+
+    decimal interval = 0.1;
+
+    taskA.scheduleTaskExecution(interval);
+    taskB.scheduleTaskExecution(interval);
+    taskC.scheduleTaskExecution(interval);
+    taskD.scheduleTaskExecution(interval);
+    taskE.scheduleTaskExecution(interval);
```
Comment (SasinduDilshara) [inline transaction-ballerina/tests/transaction_concurrency.bal:82]:
```suggestion
   decimal interval = 0.1;

    var taskA = new TaskExecuter("A");
    var taskB = new TaskExecuter("B");
    var taskC = new TaskExecuter("C");
    var taskD = new TaskExecuter("D");
    var taskE = new TaskExecuter("E");

    taskA.scheduleTaskExecution(interval);
    taskB.scheduleTaskExecution(interval);
    taskC.scheduleTaskExecution(interval);
    taskD.scheduleTaskExecution(interval);
    taskE.scheduleTaskExecution(interval);
```

**Review by gimantha — APPROVED** [review]

**Pull request merged**